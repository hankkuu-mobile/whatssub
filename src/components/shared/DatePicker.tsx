import {
  Dimensions,
  Text,
  View,
} from 'react-native';
import React, { Component, useEffect, useState } from 'react';
import Picker from 'react-native-picker-scrollview';
import { colors } from '../../theme';
import moment from 'moment';
import styled from 'styled-components/native';

const Container = styled.View`
  width: ${Dimensions.get('screen').width}px;
  height: 255px;
  background-color: ${colors.headerBackgroundLight};
  align-items: center;
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
`;

const Header = styled.View`
  align-items: center;
  border-bottom-color: #e6eaee;
  border-bottom-width: 1px;
  height: 90px;
  width: ${Dimensions.get('screen').width}px;
`;

const Title = styled.Text`
  color: ${colors.textPrimaryLight};
  font-family: 'spoqa-han-sans-bold';
  font-size: 16px;
  line-height: 19px;
  margin-top: 21px;
`;

const Description = styled.Text`
  color: ${colors.textPrimaryLight};
  font-family: 'spoqa-han-sans-regular';
  font-size: 14px;
  margin-top: 14px;
`;

const Body = styled.View`
  align-items: center;
  flex: 1;
  flex-direction: row;
  justify-content: space-around;
  padding: 0 40px;
  width: 100%;
`;

const Row = styled.View`
  align-items: center;
  flex-direction: row;
`;

const DateUnit = styled.Text`
  color: ${colors.textPrimaryLight};
  font-family: 'spoqa-han-sans-regular';
  font-size: 24px;
`;

const PickerText = styled.Text`
  background-color: ${colors.headerBackgroundLight};
  color: ${colors.blue};  
  font-family: 'spoqa-han-sans-regular';
  font-size: 24px;
  height: 35px;
`;

interface Props {
  date?: Date
  onDateChange?(newDate: Date): void
  minDate?: Date
  maxDate?: Date
  title: string
  description?: string
}

function DatePicker(props: Props) {
  const today = new Date();
  const {
    title,
    description = '날짜를 선택해주세요.',
    date = today,
    minDate = new Date(today.getFullYear() - 10, today.getMonth()),
    maxDate = new Date(today.getFullYear() + 10, today.getMonth()),
    onDateChange = () => null,
  } = props;
  const [selectedDate, setSelectedDate] = useState(date);
  useEffect(() => {
    if (onDateChange) {
      onDateChange(selectedDate);
    }
  }, [selectedDate]);
  const yearSource = Array.from(
    Array(maxDate.getFullYear() - minDate.getFullYear() + 1).keys()
  ).map((adder) => minDate.getFullYear() + adder);
  const yearIndex = selectedDate.getFullYear() - minDate.getFullYear();
  return (
    <Container>
      <Header>
        <Title>{title}</Title>
        <Description>{description}</Description>
      </Header>
      <Body>
        <Row>
          <Picker
            highlightColor={'transparent'}
            wrapperHeight={105}
            itemHeight={35}
            dataSource={yearSource}
            selectedIndex={yearIndex}
            onValueChange={(value) => setSelectedDate(
              new Date(
                value,
                selectedDate.getMonth(),
                selectedDate.getDate()
              )
            )}
            renderItem={(value, idx, isSelected) => (
              <PickerText style={{ opacity: isSelected ? 1 : 0.2 }}>{value}</PickerText>
            )}/>
          <DateUnit>년</DateUnit>
        </Row>
        <Row>
          <Picker
            highlightColor={'transparent'}
            wrapperHeight={105}
            itemHeight={35}
            dataSource={Array.from(Array(12).keys())}
            selectedIndex={selectedDate.getMonth()}
            onValueChange={(value) => setSelectedDate(
              new Date(
                selectedDate.getFullYear(),
                value,
                selectedDate.getDate()
              )
            )}
            renderItem={(value, idx, isSelected) => (
              <PickerText style={{ opacity: isSelected ? 1 : 0.2 }}>{value + 1}</PickerText>
            )}/>
          <DateUnit>월</DateUnit>
        </Row>
        <Row>
          <Picker
            highlightColor={'transparent'}
            wrapperHeight={105}
            itemHeight={35}
            dataSource={Array.from(Array(31).keys()).map((d) => d + 1)}
            selectedIndex={selectedDate.getDate() - 1}
            onValueChange={(value) => setSelectedDate(
              new Date(
                selectedDate.getFullYear(),
                selectedDate.getMonth(),
                value,
              )
            )}
            renderItem={(value, idx, isSelected) => (
              <PickerText style={{ opacity: isSelected ? 1 : 0.2 }}>{value}</PickerText>
            )}/>
          <DateUnit>일</DateUnit>
        </Row>
      </Body>
    </Container>
  );
}

export default DatePicker;
