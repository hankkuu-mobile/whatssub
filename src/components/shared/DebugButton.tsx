import React, { useContext } from 'react';
import styled, { ThemeContext } from 'styled-components/native';

import { Text } from 'react-native';

const LabelWrapper = styled.TouchableOpacity`
  height: 50;
  width: 70;
  border-radius: 200;
  position: absolute;
  bottom: 100;
  right: 20;
  justify-content: center;
  align-items: center;
  background-color: #686cc3;
`;

const DebugButton = () => {
  const { debug, setDebug } = useContext(ThemeContext);
  return (
    <LabelWrapper onPress={() => setDebug(!debug)}>
      <Text>{`Debug ${debug}`}</Text>
    </LabelWrapper>
  );
};

export default DebugButton;
