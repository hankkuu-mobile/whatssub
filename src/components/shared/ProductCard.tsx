import { IC_EDIT, IC_NOTI_OFF, IC_NOTI_ON } from '../../utils/Icons';
import React, { FC, useState } from 'react';
import { SwitchToggle } from 'dooboo-native-widgets';
import {
  TouchableOpacity,
} from 'react-native';

import { getString } from '../../../STRINGS';
import moment from 'moment';
import styled from 'styled-components/native';

const Container = styled.View<{ isSummary: boolean }>`
  display: flex;
  position: relative;
  width: 100%;
  height: ${({ isSummary }) => isSummary ? '82' : '190'};
  background-color: white;
  padding: 15px 22px;
  border-radius: 15;
  
`;

const HeaderView = styled.View<{ isSubscription: boolean }>`
  position: absolute;
  width: 100%;
  height: ${({ isSubscription }) => isSubscription ? '40' : '30'};
  top: 15;
  left: 22;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const ContentView = styled.View`
  display: flex;
`;

const SummaryContentView = styled.View`
  display: flex;
  width: 100%;
  height: 100%;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const SummaryContentViewWrapper = styled.View`
  position: absolute;
  left: 20;
  top: 30;
  width: 100%;
`;

const TitleText = styled.Text`
  font-family: 'spoqa-han-sans-bold';
  margin-top: 17;
  text-align: center;
  font-size: 16;
  font-weight: bold;
  color: ${({ theme }) => theme.fontColor};
`;

const SummaryTitleTextWrapper = styled.View`
  display: flex;
`;

const SummaryTitleText = styled.Text`
  font-family: 'spoqa-han-sans-bold';
  text-align: center;
  font-size: 16;
  font-weight: bold;
  color: ${({ theme }) => theme.fontColor};
`;

const PriceText = styled.Text`
  font-family: 'spoqa-han-sans-regular';
  font-size: 14;
  text-align: center;
  color: rgb(153, 162, 170);
  margin-top: 10;
`;

const DateText = styled.Text<{ variant: Variant, isSummary: boolean }>`
  font-family: ${({ isSummary }) =>
    isSummary ? 'spoqa-han-sans-bold' : 'spoqa-han-sans-regular'};
  font-size: ${({ isSummary }) =>
    isSummary ? '16' : '40'};
  text-align: center;
  color: ${({ variant }) =>
    variant === 'subscription'
      ? 'rgb(255, 77, 108)'
      : 'rgb(62, 126, 255)'};
`;

const ProductImage = styled.Image<{ isSubscription: boolean }>`
  width: ${({ isSubscription }) => isSubscription ? '75' : '31'};
  height: ${({ isSubscription }) => isSubscription ? '20' : '31'};
  margin-top: ${({ isSubscription }) => isSubscription ? '10' : '0'};
  margin-bottom: ${({ isSubscription }) => isSubscription ? '10' : '0'};
  margin-right: auto;
  margin-left: auto;
`;

const IconImage = styled.Image`
  width: 24;
  height: 24;
`;

const IconWrapper = styled.View`
  display: flex;
  flex-direction: row;
`;

const ContentText = styled.Text`
  font-family: 'spoqa-han-sans-bold';
  text-align: center;
  font-size: 13;  
`;

export enum Variant {
  Subscription = 'subscription',
  Payment = 'payment',
};

export interface ProductCardProps {
  onClickNotification: () => void;
  onClickSubscription?: () => void;
  onClickEdit: () => void;
  isNotificationEnable: boolean;
  isSubscribing?: boolean;
  name: string;
  image: string;
  price: number;
  currentMonthPaymentDate: Date;
  variant?: Variant;
  defaultIsSummary?: boolean;
}

const ProductCard: FC<ProductCardProps> = ({
  name, image, price, currentMonthPaymentDate, variant = Variant.Subscription,
  isNotificationEnable, isSubscribing = false, onClickSubscription, onClickNotification,
  onClickEdit, defaultIsSummary = false,
}) => {
  const dDay = moment(currentMonthPaymentDate)
    .diff(moment().startOf('day'), 'day');
  const isSubscription = variant === Variant.Subscription;
  const [isSummary, setIsSummary] = useState(defaultIsSummary);

  const onPressCard = () => {
    setIsSummary((prev) => !prev);
  };

  const onClickSwitchToggle = () => {
    onClickSubscription();
    return null;
  };

  const onPressNotiIcon = () => {
    onClickNotification();
  };

  const renderNoti = isNotificationEnable
    ? <TouchableOpacity onPress={onPressNotiIcon} testID='notiOffIcon'>
      <IconImage source={IC_NOTI_OFF}/>
    </TouchableOpacity>
    : <TouchableOpacity onPress={onPressNotiIcon} testID="notiOnIcon">
      <IconImage source={IC_NOTI_ON}/>
    </TouchableOpacity>;

  const renderSwitch = <SwitchToggle
    backTextRight={isSubscribing
      ? '' : getString('PRODUCTION_CARD_SUBSCRIPTION_SWITCH_OFF')}
    backTextLeft={isSubscribing
      ? getString('PRODUCTION_CARD_SUBSCRIPTION_SWITCH_ON') : ''}
    textRightStyle={{ color: 'white', position: 'absolute', left: 5, top: -7 }}
    textLeftStyle={{ color: 'white', position: 'absolute', left: 5, top: -7 }}
    switchOn={isSubscribing}
    backgroundColorOn={'rgb(2, 48, 89)'}
    backgroundColorOff={'rgb(2, 48, 89)'}
    circleColorOn={'white'}
    circleColorOff={'white'}
    onPress={onClickSwitchToggle}/>;

  return (
    <TouchableOpacity onPress={onPressCard} testID='productCard'>
      <Container isSummary={isSummary}>
        {isSummary
          ? <>
            <SummaryTitleTextWrapper testID='summaryTitleText'>
              <SummaryTitleText>{name}</SummaryTitleText>
              <PriceText>{price && `-${price.toLocaleString()}원`}</PriceText>
            </SummaryTitleTextWrapper>
            <SummaryContentViewWrapper>
              <SummaryContentView>
                <DateText variant={variant} isSummary={isSummary}>
                  {`D-${dDay}`}
                </DateText>
                {isSubscription ? renderSwitch : renderNoti}
              </SummaryContentView>
            </SummaryContentViewWrapper>
          </>
          : <>
            <ProductImage
              testID='productImg'
              source={{ uri: image }}
              isSubscription={isSubscription}
            />
            <HeaderView isSubscription={isSubscription}>
              <IconWrapper>
                <TouchableOpacity onPress={onClickEdit}>
                  <IconImage source={IC_EDIT}/>
                </TouchableOpacity>
                {isSubscription && renderNoti}
              </IconWrapper>
              {isSubscription ? renderSwitch : renderNoti}
            </HeaderView>
            <ContentView>
              <TitleText>{name}</TitleText>
              <DateText variant={variant} isSummary={false}>
                {`D-${dDay}`}
              </DateText>
              <ContentText>
                {`${moment(currentMonthPaymentDate).format('M월 D일')}에
                  ${price && price.toLocaleString()}원이 결제될 예정입니다.`}
              </ContentText>
            </ContentView>
          </>
        }
      </Container>
    </TouchableOpacity>
  );
};

export default ProductCard;
