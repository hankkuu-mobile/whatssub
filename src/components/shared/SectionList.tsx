import {
  SectionList as ReactNativeSectionList,
  SectionListData,
} from 'react-native';

import React from 'react';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';

const StyledSelectionList = styled(ReactNativeSectionList)`
  background-color: rgb(249, 250, 251);
`;

const SectionSeparator = styled.View`
  display: flex;
  width: 100%;
  height: 1px;
  background-color: rgb(230, 234, 238);
`;

const SectionItemSeparator = styled.View`
  display: flex;
  width: 91.47%;
  height: 1px;
  background-color: rgb(243, 245, 247);
  align-self: center;
`;

const SectionEmptyHeader = styled.View`
  display: flex;
  justify-content: center;
  height: 12px;
  background-color: rgb(249, 250, 251);

  padding: 0 16px;
`;

const SectionHeader = styled(SectionEmptyHeader)`
  height: 36px;
`;

const SectionHeaderText = styled.Text`
  font-size: 12;
  color: rgb(153, 162, 170);
  letter-spacing: -0.5;
`;

interface Props {
  sections: readonly SectionListData<any>[];
  localizeTitle?: boolean;
  [prop: string]: any;
}

function SectionList({ sections, localizeTitle, ...props }: Props) {
  return (
    <StyledSelectionList
      renderSectionHeader={({ section: { title } }) => (
        <>
          <SectionSeparator />
          {title ? (
            <SectionHeader testID='sectionHeader'>
              <SectionHeaderText>
                {localizeTitle ? getString(title) : title}
              </SectionHeaderText>
            </SectionHeader>
          ) : (
            <SectionEmptyHeader testID='sectionEmptyHeader' />
          )}
        </>
      )}
      sections={sections}
      ItemSeparatorComponent={SectionItemSeparator}
      ListFooterComponent={SectionSeparator}
      keyExtractor={(item, index) => item + index}
      {...props}
    />
  );
}

export default SectionList;
