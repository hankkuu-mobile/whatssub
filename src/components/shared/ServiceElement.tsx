import PropTypes from 'prop-types';
import React from 'react';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';

/**
This component is being used at SearchService screen.
Showing service.

Orientation behaviors
- ImageContainer will be stretched/shrinks

By splitting containers and contents(image, name, category),
- it has better readability and
- its container design can be modified/reused easily

Simple container explanation
- Container - top (contain below three containers)
- ImageContainer - left most, fixed size
- ServiceContainer - stretch/shrinks
- NameContainer - right most, fixed size

To reuse, you may want to refactor its containers out in the future

To debug its layout manually, you may put background-color to each Containers easily
 */

const Container = styled.View`
  background-color: white;
  flex-direction: row;
  align-items: stretch;
  height: 70;
`;
const ImageContainer = styled.View`
  flex-basis: 78;
  flex-grow: 0;
  justify-content: center;
  align-items: center;
`;
const ServiceImage = styled.Image`
  width: 42;
  height: 42;
  border-radius: 5;
`;
const NameContainer = styled.View`
  flex-grow: 1;
  justify-content: center;
`;
const Name = styled.Text`
  font-size: 16;
  color: #323b43;
  letter-spacing: -0.5;
`;
const CategoryContainer = styled.View`
  flex-basis: 108;
  flex-grow: 0;
  justify-content: center;
  padding-right: 24;
`;
const Category = styled.Text`
  text-align: right;
  font-size: 14;
  color: #99a2aa;
`;

const ServiceElement = ({ imageSource, name, category }) => {
  return (
    <Container>
      <ImageContainer>
        <ServiceImage source={imageSource} />
      </ImageContainer>
      <NameContainer>
        <Name testID='SERVICE_ELEMENT_NAME'>{name}</Name>
      </NameContainer>
      <CategoryContainer>
        <Category testID='SERVICE_ELEMENT_CATEGORY'>{category}</Category>
      </CategoryContainer>
    </Container>
  );
};

ServiceElement.propTypes = {
  imageSource: PropTypes.shape({
    uri: PropTypes.string.isRequired,
  }).isRequired,
  name: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
};

export default ServiceElement;
