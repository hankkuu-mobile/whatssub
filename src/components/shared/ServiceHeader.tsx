import {
  Dimensions,
  Image,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  View } from 'react-native';
import React, { Component } from 'react';

import { IC_WRITE } from '../../utils/Icons';
import { IMG_MAIN } from '../../utils/Images';
import { colors } from '../../theme';
import styled from 'styled-components/native';

const Container = styled.View`
  width: 100%;
  background-color: ${colors.headerBackgroundLight};
  align-items: center;
  justify-content: center;
`;

const ImageContainer = styled.View`
  height: 86px;
  align-items: center;
  justify-content: center;
`;

const MainImage = styled.Image`
  max-height: 86px;
  width: ${Dimensions.get('window').width}px;
`;

const ServiceName = styled.Text`  
  font-family: 'spoqa-han-sans-bold';
  font-size: 16px;
`;

const ServiceNameInput = styled.TextInput`  
  font-family: 'spoqa-han-sans-bold';
  font-size: 16px;
`;

const ServiceDescription = styled.Text`  
  font-family: 'spoqa-han-sans-regular';
  font-size: 14px;
  color: ${colors.gray};
  margin-top: 13px;
  margin-bottom: 20px;
`;

interface Props {
  image?: number
  custom?: boolean
  serviceName?: string
  description?: string
  onChangeServiceName?(string): void
}

function Shared(props: Props) {
  const {
    image = IMG_MAIN,
    description = '서비스에 대한 요금과 결제정보를 입력해주세요.',
    custom = false,
    serviceName,
  } = props;
  const titleRef = React.useRef(null);
  return (
    <Container>
      <ImageContainer>
        <MainImage source={image} resizeMode='contain' />
      </ImageContainer>
      <View style={{ flexDirection: 'row', alignItems: 'center' }}>
        {
          custom
            ? <ServiceNameInput
              ref={titleRef}
              placeholder='서비스명을 직접 입력해주세요.'
              value={serviceName}
              placeholderTextColor='black'
              onChangeText={props.onChangeServiceName}/>
            : <ServiceName>{serviceName}</ServiceName>
        }
        {
          custom
            ? <TouchableOpacity onPress={() => titleRef != null && titleRef.current.focus()}>
              <Image source={IC_WRITE} style={{ width: 14, height: 16, marginHorizontal: 4 }} />
            </TouchableOpacity>
            : null
        }
      </View>
      <ServiceDescription>{description}</ServiceDescription>
    </Container>
  );
}

export default Shared;
