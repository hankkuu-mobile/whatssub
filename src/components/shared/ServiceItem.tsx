import React, { Component, useRef } from 'react';
import { TouchableOpacity, View } from 'react-native';

import { IC_LIST_BTN } from '../../utils/Icons';
import { colors } from '../../theme';
import styled from 'styled-components/native';

const Container = styled.View`
  height: 50px;
  width: 100%;
  background-color: ${colors.headerBackgroundLight};
  flex-direction: row;
  align-items: center;
  padding: 0 16px;
`;

const Title = styled.Text`
  color: ${colors.textPrimaryLight};
  font-size: 16px;
  font-family: 'spoqa-han-sans-regular';
`;

const Content = styled.TextInput`
  color: ${colors.textSecondaryLight};
  font-size: 16px;
  font-family: 'spoqa-han-sans-regular';
`;

const Arrow = styled.Image`
  width: 8px;
  height: 13px;
  margin-left: 11px;
`;

interface Props {
  title: string;
  content: string;
  testID?: string;
  contentStyle?: any;
  showArrow?: boolean;
  onPress?(): any;
  editable?: boolean;
  onChangeContent?(string): void;
}

function ServiceItem(props: Props) {
  const {
    title = '제목',
    content = '내용',
    contentStyle,
    showArrow,
    editable = false,
    onPress = null,
    onChangeContent = null,
  } = props;
  const inputText = useRef(null);
  return (
    <TouchableOpacity
      testID={props.testID}
      onPress={
        // prettier-ignore
        props.onPress
          ? props.onPress
          : editable && inputText != null
            ? () => inputText.current.focus()
            : null
      }
      activeOpacity={props.onPress ? 0.2 : 1}
    >
      <Container>
        <Title>{title}</Title>
        <View style={{ flex: 1 }} />
        <Content
          ref={inputText}
          style={contentStyle}
          editable={editable}
          pointerEvents={editable ? 'auto' : 'none'}
          onChangeText={onChangeContent}
        >
          {content}
        </Content>
        {showArrow ? <Arrow source={IC_LIST_BTN} /> : null}
      </Container>
    </TouchableOpacity>
  );
}

export default ServiceItem;
