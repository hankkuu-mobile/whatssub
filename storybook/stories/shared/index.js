import './Button';
import './SectionList';
import './DebugButton';
import './PieChart';
import './ProductCard';
import './ServiceHeader';
import './ServiceItem';
import './DatePicker';
import './ServiceElement';
