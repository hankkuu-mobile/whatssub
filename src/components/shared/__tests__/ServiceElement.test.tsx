import 'react-native';

import React from 'react';
import { SERVICE_ELEMENT_MOCK } from '../../../../storybook/stories/shared/ServiceElement';
import ServiceElement from '../ServiceElement';
import renderer from 'react-test-renderer';

describe('[ServiceElement]', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ServiceElement {...SERVICE_ELEMENT_MOCK} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
