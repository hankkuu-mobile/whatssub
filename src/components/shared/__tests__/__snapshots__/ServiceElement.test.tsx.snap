// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ServiceElement] renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 70,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 78,
          "flexGrow": 0,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://facebook.github.io/react-native/docs/assets/favicon.png",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 5,
            "height": 42,
            "width": 42,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexGrow": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#323b43",
            "fontSize": 16,
            "letterSpacing": -0.5,
          },
        ]
      }
      testID="SERVICE_ELEMENT_NAME"
    >
      name
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 108,
          "flexGrow": 0,
          "justifyContent": "center",
          "paddingRight": 24,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#99a2aa",
            "fontSize": 14,
            "textAlign": "right",
          },
        ]
      }
      testID="SERVICE_ELEMENT_CATEGORY"
    >
      category
    </Text>
  </View>
</View>
`;
